### DISCUSSION TITLE
editorconfig: unofficial `max_line_length` key is supported

### DISCUSSION THREAD

**Discussion by muzimuzhi** [biomejs/biome#4762] [General]
Biome supports `max_line_width` key used in `.editorconfig` (https://github.com/biomejs/biome/commit/78d70d2a075fee32d3694206ff4a6597aa551117), but actually `max_line_width` is [not officially supported](https://spec.editorconfig.org/#supported-pairs) by EditorConfig. Maybe Biome should drop support for it.

The commit message of https://github.com/benesch/materialize/commit/31f2d7f59f25639c22d12f4fced961c9e045bce6 can be used as a good summary:

> The max_line_length directive is not officially part of the EditorConfig
spec, and its current implementations have been contentious [[0]] [[1]]. The
tl;dr is that it is not clear whether to use max_line_length for soft
wrapping, hard wrapping, or just a visual display of a ruler, and
various tools and editors have different opinions.

[0]: https://github.com/editorconfig/editorconfig/issues/280
[1]: https://github.com/editorconfig/editorconfig/issues/387

Related discussions in editorconfig repos:
- [Property research: maximum line length](https://github.com/editorconfig/editorconfig/wiki/Property-research:-maximum-line-length) | EditorConfig wiki
- https://github.com/editorconfig/editorconfig/issues/89
- https://github.com/editorconfig/editorconfig-vscode/issues/53
- https://github.com/editorconfig/editorconfig/issues/387

**Comment by ematipico** [comment]
Thank you for letting us know. Tracked here https://github.com/biomejs/biome/issues/4892
